### ISSUE TITLE
Why does the init method need to be delayed by 1 second before execution? 

### ISSUE THREAD

**Issue by fstonezst** [yfinkelstein/node-zookeeper#352]
https://github.com/yfinkelstein/node-zookeeper/blob/cebae9cc39f4f1f8c1d072d364938e6cbfbb8067/examples/wrapper.js#L59C1-L60C1

I have two questions regarding the code mentioned above. Could you please assist me with them? Thank you!
1. Why does the init method need to be delayed by 1 second before execution? 
2. Shouldn't the parameter for the init method be config?

**Comment by DavidVujic** [comment]
You're right: the `config` could be passed in to the `init()` function. The configuration (also known as `options`) is possible to add in the constructor, or the init method. It's a feature that has evolved over time and both are still there to not cause any breaking changes for users. This library has been around for many years.

About the waiting, it's mainly to make the server get a chance to hook up, and to make the call asynchronous.

**Comment by fstonezst** [comment]
> You're right: the `config` could be passed in to the `init()` function. The configuration (also known as `options`) is possible to add in the constructor, or the init method. It's a feature that has evolved over time and both are still there to not cause any breaking changes for users. This library has been around for many years.
> 
> About the waiting, it's mainly to make the client and server get a chance to hook up, and to make the call asynchronous/non-blocking.

Thank you for the clarification regarding the `init` parameters; I understand now. However, I'm still a bit confused about the waiting aspect. Can I use `client.init(config);` to replace the following code?

```javascript
setTimeout(() => {
    client.init({});
    startTimer();
}, 1000);
```

Or could I move the `init` call into the `connect` event instead?

**Comment by DavidVujic** [comment]
I think you could do without the setTimeout (even though it is recommended). 

But you can't move it into the `connect` event, because that won't happen until after `init`. The constructor only creates a client, but not a connection.

**Comment by fstonezst** [comment]
> I think you could do without the setTimeout (even though it is recommended).
> 
> But you can't move it into the `connect` event, because that won't happen until after `init`. The constructor only creates a client, but not a connection.

I understand your point now. Thank you very much for your support! Wishing you all the best in your endeavors!
